tests/src/main/java/net/neoforged/neoforge/debug/effect/MobEffectTests.java

```diff
@@ -0,0 +1,91 @@
+                MobEffectCategory.HARMFUL, 0xFF0000) {
+            @Override
+            public void fillEffectCures(Set<EffectCure> cures, MobEffectInstance effectInstance) {
+                super.fillEffectCures(cures, effectInstance);
+                cures.remove(EffectCures.MILK);
+                cures.add(testCureTwo);
+            }
+        });
+
+        test.eventListeners().forge().addListener((MobEffectEvent.Added event) -> {
+            if (event.getEffectInstance().getEffect() == MobEffects.NIGHT_VISION) {
+                event.getEffectInstance().getCures().add(testCure);
+            }
+        });
+
+        test.onGameTest(helper -> {
+            Pig pig = helper.spawnWithNoFreeWill(EntityType.PIG, 1, 1, 1);
+
+            pig.addEffect(new MobEffectInstance(MobEffects.CONFUSION));
+            helper.assertMobEffectPresent(pig, MobEffects.CONFUSION, "'confusion was applied'");
+            pig.removeEffectsCuredBy(testCure);
+            helper.assertMobEffectPresent(pig, MobEffects.CONFUSION, "'confusion not removed by test cure'");
+            pig.removeEffectsCuredBy(EffectCures.MILK);
+            helper.assertMobEffectAbsent(pig, MobEffects.CONFUSION, "'confusion removed by milk'");
+
+            pig.addEffect(new MobEffectInstance(MobEffects.NIGHT_VISION));
+            helper.assertMobEffectPresent(pig, MobEffects.NIGHT_VISION, "'nightvision was applied'");
+            pig.removeEffectsCuredBy(testCure);
+            helper.assertMobEffectAbsent(pig, MobEffects.NIGHT_VISION, "'nightvision removed by test cure'");
+
+            pig.addEffect(new MobEffectInstance(testEffect.get()));
+            helper.assertMobEffectPresent(pig, testEffect.get(), "'test effect was applied'");
+            pig.removeEffectsCuredBy(EffectCures.MILK);
+            helper.assertMobEffectPresent(pig, testEffect.get(), "'test effect not removed by milk'");
+            pig.removeEffectsCuredBy(testCureTwo);
+            helper.assertMobEffectAbsent(pig, testEffect.get(), "'test effect removed by test cure'");
+
+            MobEffectInstance srcInst = new MobEffectInstance(MobEffects.CONFUSION);
+            MobEffectInstance destInst = MobEffectInstance.load(srcInst.save(new CompoundTag()));
+            helper.assertTrue(srcInst.getCures().equals(destInst.getCures()), "'MobEffectInstance serialization roundtrip (standard cures)'");
+
+            srcInst.getCures().add(testCure);
+            destInst = MobEffectInstance.load(srcInst.save(new CompoundTag()));
+            helper.assertTrue(srcInst.getCures().equals(destInst.getCures()), "'MobEffectInstance serialization roundtrip (custom additional cure)'");
+
+            srcInst.getCures().clear();
+            destInst = MobEffectInstance.load(srcInst.save(new CompoundTag()));
+            helper.assertTrue(srcInst.getCures().equals(destInst.getCures()), "'MobEffectInstance serialization roundtrip (no cures)'");
```

Comment:
Why are you surrounding every message in quotes?

Reply:
To me `Entity Pig[...] failed 'do the thing' test` reads better than `Entity Pig[...] failed do the thing test`. I'm not gonna die on that hill though